core/src/main/java/io/parsingdata/metal/data/ImmutableList.java

```diff
@@ -59,9 +63,13 @@ private ImmutableList(final T head, final ImmutableList<T> tail) {
 
     public ImmutableList<T> add(final ImmutableList<T> list) {
         checkNotNull(list, "list");
-        if (list.isEmpty()) { return this; }
```

Comment:
But what if the added check is more expensive? CPU don't like branching :laughing: